bundles/org.eclipse.e4.ui.css.swt.theme/src/org/eclipse/e4/ui/css/swt/internal/theme/Theme.java

```diff
@@ -43,10 +44,19 @@ public String getOsVersion() {
 		return this.osVersion;
 	}
 
+	@Override
+	public boolean isDark() {
+		return Boolean.parseBoolean(isDark);
```

Comment:
for older theme contributions that have not yet updated to the explicit isDark attribute, we should probably also fall back to the heuristic here to maximize compatibility, otherwise, all legacy themes would suddenly become light.

Reply:
> for older theme contributions that have not yet updated to the explicit isDark attribute, we should probably also fall back to the heuristic here to maximize compatibility, otherwise, all legacy themes would suddenly become light.

And when will we get rid of that fallback?

Reply:
Hmm, for a few releases?

We should announce this in the new-and-noteworthy, that themes may now be explicitly set to dark/light and therefore there is no need to have `...dark...` in the ID anymore.

Additionally, consumers are required to explicitly set this ID as we want to remove the fallback with release X.

X being derived from the same rules as when we mark API as `forRemoval`.

Reply:
see 8fe3319